python/foglamp/services/core/service_registry/monitor.py

```diff
@@ -31,60 +30,61 @@ class Monitor(object):
 
     async def _sleep(self, sleep_time):
         await asyncio.sleep(sleep_time)
 
     async def _monitor_loop(self):
         """async Monitor loop to monitor registered services"""
         # check health of all micro-services every N seconds
-        _MAX_ATTEMPTS = 15
-        """Number of max attempts for finding a heartbeat of service"""
 
         while True:
+            self._logger.info("Starting next round of service monitoring, sleep/i:{} ping/t:{} max/a:{}".format(
+                self._sleep_interval, self._ping_timeout, self._max_attempts))
             for service_record in ServiceRegistry.all():
-                attempt_count = 1
-                """Number of current attempt to ping url"""
-
-                url = "{}://{}:{}/foglamp/service/ping".format(service_record._protocol, service_record._address,
-                                                               service_record._management_port)
-                async with aiohttp.ClientSession() as session:
-                    while attempt_count < _MAX_ATTEMPTS + 1:
-                        # self._logger.info("Attempting service %s with try count %d", url, attempt_count)
-                        try:
+                # No need to try ping if service status is either Unregistered or Failed
+                if service_record._status in [ServiceRecord.Status.Running, ServiceRecord.Status.Doubtful]:
+                    try:
+                        url = "{}://{}:{}/foglamp/service/ping".format(
+                            service_record._protocol, service_record._address, service_record._management_port)
+                        async with aiohttp.ClientSession() as session:
                             async with session.get(url, timeout=self._ping_timeout) as resp:
                                 text = await resp.text()
                                 res = json.loads(text)
                                 if res["uptime"] is None:
                                     raise ValueError('Improper Response')
-                                break
-                        # TODO: Fix too broad exception clause
-                        except:
-                            attempt_count += 1
-                            await self._sleep(1.5)
-                    if attempt_count > _MAX_ATTEMPTS:
-                        service_record._status = 0
+                    except:  # TODO: Fix too broad exception clause
+                        service_record._status = ServiceRecord.Status.Doubtful
+                        service_record._check_count += 1
+                        self._logger.info("Marked as doubtful micro-service %s", service_record.__repr__())
+                    else:
+                        service_record._status = ServiceRecord.Status.Running
+                        service_record._check_count = 1
+
+                    if service_record._check_count > self._max_attempts:
+                        service_record._status = ServiceRecord.Status.Failed
```

Comment:
If we unregister then this will take the service out of the service registered and hence it will not show up as failed. Unregister is really now only used when the service does a graceful shutdown.